pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -681,12 +681,35 @@ pub mod pallet {
 					sum.saturating_add(T::Fungibles::total_issuance(id))
 				});
 
-			let amount = burnt
-				.checked_mul(&total_collateral_issuance)
-				.ok_or(ArithmeticError::Overflow)? // TODO: do we need a fallback if this fails?
+			defensive_assert!(
+				sum_of_issuances >= burnt,
+				"burnt amount exceeds the total supply of all bonded currencies"
+			);
+
+			// divide first to avoid overflow, remainder will be handled seperately
+			let divisible_part = total_collateral_issuance
```

Comment:
I realised that the approach of handling remainders separately can also improve the calculations of how much collateral is refunded, preventing overflow in many (but not all) cases.